selfdrive/car/subaru/subarucan.py

```diff
@@ -1,19 +1,36 @@
 import copy
 from cereal import car
+from selfdrive.car.subaru.values import CAR
 
 VisualAlert = car.CarControl.HUDControl.VisualAlert
 
-def create_steering_control(packer, apply_steer, frame, steer_step):
+def subaru_preglobal_checksum(packer, values, addr):
+  dat = packer.make_can_msg(addr, 0, values)[2]
+  return (sum(dat[:7])) % 256
 
-  # counts from 0 to 15 then back to 0 + 16 for enable bit
-  idx = ((frame // steer_step) % 16)
+def create_steering_control(packer, car_fingerprint, apply_steer, frame, steer_step):
 
-  values = {
-    "Counter": idx,
-    "LKAS_Output": apply_steer,
-    "LKAS_Request": 1 if apply_steer != 0 else 0,
-    "SET_1": 1
-  }
+  if car_fingerprint in [CAR.ASCENT, CAR.FORESTER, CAR.IMPREZA]:
+    #counts from 0 to 15 then back to 0 + 16 for enable bit
+    idx = (frame / steer_step) % 16
+
+    values = {
+      "Counter": idx,
+      "LKAS_Output": apply_steer,
+      "LKAS_Request": 1 if apply_steer != 0 else 0,
+      "SET_1": 1
+    }
+
+  if car_fingerprint == CAR.LEGACY_2015:
```

Comment:
I'd just make a list of all the legacy cars in values.py and check if the car's in the list.

Reply:
done